lib/absinthe/schema/notation.ex

```diff
@@ -48,6 +48,94 @@ defmodule Absinthe.Schema.Notation do
     scope(__CALLER__, :object, identifier, [], block)
   end
 
+  @placement {:object, [toplevel: true]}
+  defmacro fast_object(identifier, attrs, [do: block]) do
+    fast_scope(__CALLER__, :object, identifier, attrs, block)
+  end
+  defmacro fast_object(identifier, [do: block]) do
+    fast_scope(__CALLER__, :object, identifier, [], block)
+  end
+
+  @doc false
+  # Define a notation scope that will accept attributes
+  def fast_scope(env, kind, identifier, attrs, block) do
+    fast_open_scope(kind, env, identifier, attrs)
+
+    block
+    |> expand(env)
```

Comment:
expansion effectively evaluates the field calls, meaning we can just build the scope now instead of putting stuff in the module body that builds scope later.